invenio_i18n/collections/collect.py

```diff
@@ -0,0 +1,192 @@
+# Copyright (C) 2025 Graz University of Technology.
+#
+# Invenio is free software; you can redistribute it and it
+# under the terms of the MIT License; see LICENSE file for more details.
+"""Collect PO translations, convert to JSON, and validate."""
+
+from __future__ import annotations
+
+from pathlib import Path
+
+import polib
+
+from .convert import po_to_i18next_json
+from .discovery import (
+    find_package_path,
+    iter_po_files,
+    normalize_package_to_module_name,
+)
+from .io import write_json_file
+from .validate import validate_po
+
+
+def scan_package_for_translations(package_name: str) -> dict[str, dict[str, str]]:
+    """Get all translations from one package.
+
+    :param package_name: Name of the package like 'invenio-app-rdm'
+    :return: Translations organized by language like {"de": {...}, "fr": {...}}
+    """
+
+    package_root = find_package_path(package_name)
+    translations_by_locale = {}
+
+    if not package_root:
+        return translations_by_locale
+
+    for locale, po_path in iter_po_files(package_root, package_name):
+        pofile = polib.pofile(str(po_path))
+        translations_by_locale[locale] = po_to_i18next_json(pofile, package_name)
+
+    return translations_by_locale
+
+
+def scan_package_for_validation(package_name: str) -> list[dict]:
+    """Check one package for translation problems.
+
+    :param package_name: Name of the package to check like 'invenio-app-rdm'
+    :return: List of reports showing what needs to be fixed
+    """
+
+    package_root = find_package_path(package_name)
+    validation_reports = []
+
+    if not package_root:
+        return validation_reports
+
+    for locale, po_path in iter_po_files(package_root, package_name):
+        pofile = polib.pofile(str(po_path))
+        validation_reports.append(validate_po(pofile, package_name, locale, po_path))
+
+    return validation_reports
+
+
+def collect_translations_to_json(
+    packages: list[str],
+    output_dir: Path,
+) -> dict:
+    """Collect translations from packages and save as JSON files.
+
+    :param packages: List of package names like ['invenio-app-rdm', 'invenio-rdm-records']
+    :param output_dir: Where to save the translation files
+    :return: Summary with number of packages and languages processed
+    """
+
+    results: dict[str, dict[str, dict[str, str]]] = {}
+
+    for package_name in packages:
+        translations_by_locale = scan_package_for_translations(package_name)
+
+        # Write per-package JSON under translations/<package>/translations.json
+        if translations_by_locale:
+            normalized_name = normalize_package_to_module_name(package_name)
+            package_output = output_dir / normalized_name / "translations.json"
+            write_json_file(package_output, translations_by_locale)
+
+            # Store for merged output
+            for locale, translations in translations_by_locale.items():
+                if locale not in results:
```

Comment:
nit: you could check out `collections.defaultdict` to overcome the `if locale not in results` pattern. you could write then:

```
results = defaultdict(dict)
results[locale][normalized_name] = translations
```
since results is a default dict `results[locale]` would automatically create a dict, if it is not set yet.

sidenote: i didn't tested it yet with type hints.

Reply:
yes, this also creates empty dict for accessing new keys. But in this case i will simplify the function due to type hints..